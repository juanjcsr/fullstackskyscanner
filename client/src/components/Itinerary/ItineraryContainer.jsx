import React, { Component } from 'react';
import BpkButton from 'bpk-component-button';
import { BpkGridContainer, BpkGridRow, BpkGridColumn } from 'bpk-component-grid';
import { BpkSpinner } from 'bpk-component-spinner';

import { getSearch, getSearchByPage } from './../../utils/api';
import Itinerary from './Itinerary';
import SearchForm from './../SearchForm/SearchForm';
import ItineraryHeader from './ItineraryHeader';
import ItineraryFilter from './ItineraryFilter';
import STYLES from './Itinerary.scss';

const c = className => STYLES[className] || 'UNKNOWN';


class ItineraryContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      itineraries: [],
      query: null,
      currency: {},
      session: '',
      currentPage: 0,
      pending: true,
      loading: false,
    };

    // const api = new ApiUtils();
    this.getItineraries = this.getItineraries.bind(this);
    this.onSeeMoreClick = this.onSeeMoreClick.bind(this);
    this.getFligthData = this.getFligthData.bind(this);
    this.onSearchAgainClick = this.onSearchAgainClick.bind(this);
  }

  componentDidMount() {
    console.log('mounted');
    // this.getItineraries()
  }

  onSeeMoreClick() {
    const nextPage = this.state.currentPage + 1;
    this.getItineraries(nextPage);
  }

  onSearchAgainClick() {
    this.setState({
      itineraries: [],
      query: null,
    });
  }

  getFligthData(values) {
    this.getItineraries({ ...values });
  }

  getItineraries(params) {
    this.setState({ loading: true });
    return getSearchByPage(params).then((d) => {
      this.setState({
        itineraries: d.itineraries,
        currency: d.currencies,
        agents: d.agents,
        query: d.query,
        loading: false,
      });
    });
  }

  render() {
    const {
      itineraries,
      currency,
      query,
      agents,
    } = this.state;
    const CustomSeeMore = () => (
      <div>
        <BpkButton onClick={this.onSeeMoreClick}>See More</BpkButton>
      </div>
    );

    const CustomSearchAgain = () => (
      <div>
        <BpkButton onClick={this.onSearchAgainClick}>Search Again</BpkButton>
      </div>
    );
    return (
      <div>
        <BpkGridContainer>
          {
            query == null &&
            <BpkGridRow>
              <SearchForm onSubmit={this.getFligthData} />
            </BpkGridRow>
          }
          {
            query != null &&
            <BpkGridRow>
              <BpkGridColumn width={12}>
                <ItineraryHeader header={query} />
                <ItineraryFilter className={c('ItineraryFilter')} />
              </BpkGridColumn>
            </BpkGridRow>
          }
          {
            itineraries.length > 0 &&
            <BpkGridRow>
              <BpkGridColumn width={12}>
                <Itinerary itineraries={itineraries} currency={currency} agents={agents} />
                <CustomSeeMore />
              </BpkGridColumn>
            </BpkGridRow>
          }
          {
            query != null &&
            <BpkGridRow>
              <BpkGridColumn width={12}>
                <CustomSearchAgain />
              </BpkGridColumn>
            </BpkGridRow>
          }
          {
            this.state.loading &&
            <BpkSpinner />
          }
        </BpkGridContainer>
      </div>
    );
  }
}

export default ItineraryContainer;
