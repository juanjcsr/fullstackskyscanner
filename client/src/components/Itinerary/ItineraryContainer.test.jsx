import React from 'react';
import ReactDOM from 'react-dom';

import TestRenderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import ItineraryContainer from './ItineraryContainer'

Enzyme.configure({ adapter: new Adapter() });

describe('ItineraryContainer', () => {
  it('should render correctly without flight data', () => {
    const component = TestRenderer.create(<ItineraryContainer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
