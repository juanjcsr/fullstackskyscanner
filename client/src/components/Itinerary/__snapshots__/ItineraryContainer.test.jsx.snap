// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryContainer should render correctly without flight data 1`] = `
<div>
  <div
    className="bpk-grid__container"
  >
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <form
        onSubmit={[Function]}
      >
        <label
          className="bpk-label"
        >
          From: 
        </label>
        <input
          aria-invalid={false}
          className="bpk-input bpk-input--docked-first"
          disabled={true}
          id="from_text"
          name="from"
          placeholder="Country, city or airport"
          type="text"
          value="Edimburg"
        />
        <label
          className="bpk-label"
        >
          To: 
        </label>
        <input
          aria-invalid={false}
          className="bpk-input bpk-input--docked-last"
          disabled={true}
          id="to_text"
          name="to"
          placeholder="Country, city or airport"
          type="text"
          value="London"
        />
        <label
          className="bpk-label"
        >
          Number of adults
        </label>
        <select
          aria-invalid={false}
          className="bpk-select"
          disabled={false}
          id="adults"
          label="Number of adults"
          name="adults"
          onChange={[Function]}
          value="1"
        >
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="4"
          >
            5
          </option>
          <option
            value="5"
          >
            6
          </option>
          <option
            value="6"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
        </select>
        <label
          className="bpk-label"
        >
          Number of children
        </label>
        <select
          aria-invalid={false}
          className="bpk-select"
          disabled={false}
          id="children"
          label="Number of children"
          name="children"
          onChange={[Function]}
          value="0"
        >
          <option
            value="8"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="4"
          >
            5
          </option>
          <option
            value="5"
          >
            6
          </option>
          <option
            value="6"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
        </select>
        <label
          className="bpk-label"
        >
          Number of infants
        </label>
        <select
          aria-invalid={false}
          className="bpk-select"
          disabled={false}
          id="infants"
          label="Number of infants"
          name="infants"
          onChange={[Function]}
          value="0"
        >
          <option
            value="8"
          >
            0
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="4"
          >
            5
          </option>
          <option
            value="5"
          >
            6
          </option>
          <option
            value="6"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
        </select>
        <label
          className="bpk-label"
        >
          Class
        </label>
        <select
          aria-invalid={false}
          className="bpk-select"
          disabled={false}
          id="fclass"
          label="Flight Class"
          name="fclass"
          onChange={[Function]}
          value="economy"
        >
          <option
            value="economy"
          >
            Economy
          </option>
          <option
            value="premiumeconomy"
          >
            Premium Economy
          </option>
          <option
            value="business"
          >
            Business
          </option>
          <option
            value="first"
          >
            First
          </option>
        </select>
        <label
          className="bpk-label"
        >
          Departure date: 
        </label>
        <input
          aria-invalid={false}
          className="bpk-input bpk-input--docked-first"
          disabled={true}
          id="departure_text"
          name="departure"
          placeholder="Country, city or airport"
          type="text"
          value="2019-03-25"
        />
        <label
          className="bpk-label"
        >
          Return Date: 
        </label>
        <input
          aria-invalid={false}
          className="bpk-input bpk-input--docked-last"
          disabled={true}
          id="return_text"
          name="returnd"
          placeholder="Country, city or airport"
          type="text"
          value="2019-03-26"
        />
        <label
          className="bpk-label"
        >
          Search type:
        </label>
        <select
          aria-invalid={false}
          className="bpk-select"
          disabled={false}
          id="searchType"
          label="Search type"
          name="searchType"
          onChange={[Function]}
          value="paginated"
        >
          <option
            value="paginated"
          >
            Paginated
          </option>
          <option
            value="all"
          >
            All results (slower)
          </option>
        </select>
        <button
          className="bpk-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search Flights
        </button>
      </form>
    </div>
  </div>
</div>
`;
